frontend/src/pages/pipelines/global/pipelines/PipelinesPageHeaderActions.tsx

```diff
@@ -1,39 +1,72 @@
 import * as React from 'react';
-import { Dropdown, DropdownItem, DropdownSeparator, DropdownToggle } from '@patternfly/react-core';
+import {
+  Dropdown,
+  DropdownItem,
+  DropdownSeparator,
+  MenuToggle,
+  Tooltip,
+} from '@patternfly/react-core';
 import { DeleteServerModal, ViewServerModal } from '~/concepts/pipelines/context';
 
-const PipelinesPageHeaderActions: React.FC = () => {
+type PipelinesPageHeaderActionsProps = {
+  toggleElement: React.ReactNode;
+  variant: 'plain' | 'secondary';
+  isDisabled: boolean;
+};
+
+const PipelinesPageHeaderActions: React.FC<PipelinesPageHeaderActionsProps> = ({
+  toggleElement,
+  variant,
+  isDisabled,
+}) => {
   const [open, setOpen] = React.useState(false);
   const [deleteOpen, setDeleteOpen] = React.useState(false);
   const [viewOpen, setViewOpen] = React.useState(false);
+  const onToggleClick = () => {
+    setOpen(!open);
+  };
 
   return (
     <>
-      <Dropdown
-        removeFindDomNode
-        onSelect={() => setOpen(false)}
-        toggle={
-          <DropdownToggle toggleVariant="secondary" onToggle={() => setOpen(!open)}>
-            Pipeline server actions
-          </DropdownToggle>
+      <Tooltip
+        content={
+          isDisabled
+            ? 'To access the pipeline server actions, first create a pipeline server.'
+            : 'Pipeline Server Actions'
         }
-        isOpen={open}
-        position="right"
-        dropdownItems={[
-          <DropdownItem key="view-server-details" onClick={() => setViewOpen(true)}>
-            View pipeline server configuration
-          </DropdownItem>,
-          <DropdownSeparator key="separator" />,
-          <DropdownItem
-            onClick={() => {
-              setDeleteOpen(true);
-            }}
-            key="delete-server"
-          >
-            Delete pipeline server
-          </DropdownItem>,
-        ]}
-      />
+        position="bottom"
+      >
+        <Dropdown
+          removeFindDomNode
+          onSelect={() => setOpen(false)}
+          toggle={
+            <MenuToggle
```

Comment:
Use `DropdownToggle` because keyboard navigation to the menu items does not work when you mix `Dropdown` with `MenuToggle`.

Reply:
I've replaced the `DropdownToggle` with `MenuToggle` as I'm reusing the `PipelinePageHeaderActions` to render both a Dropdown and a Kebab(as per https://www.patternfly.org/components/menus/dropdown#with-kebab-toggle) as both the menus contain the same dropdown items.

If I use `DropdownToggle`, the kebab menu looks something like this:
<img width="67" alt="Screenshot 2023-09-13 at 6 47 02 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/22885912/6dfaa41d-1349-4569-bb6b-c9f0786405fa">

Reply:
@manaswinidas I think PF5 has much better support for `Menu`. In PF4 a [dropdown with a kebab](http://v4-archive.patternfly.org/v4/components/dropdown#kebab-toggle) uses `KebabToggle`.

Reply:
This would require me to pass down `KebabToggle` from `PipelineSection` and `DropdownToggle` from `GlobalPipelines` to the `toggle` prop of `Dropdown` in `PipelinesPageHeaderActions` and handle toggle actions from `PipelinesSection` and `GlobalPipelines`. Should I go ahead?

Reply:
Do you really need to pass everything down or simply define your own variant prop and switch which toggle you render based on it?

Reply:
Done. Marking this as Resolved.